tests/unittest_inference.py

```diff
@@ -6797,5 +6797,113 @@ def test_function_def_cached_generator() -> None:
+        [
+            pytest.param(
+                """"My name is %s, I'm %s" % ("Daniel", 12)""", id="empty-indexes"
+            ),
+            pytest.param(
+                """"My name is %0s, I'm %1s" % ("Daniel", 12)""",
+                id="numbered-indexes",
+            ),
+            pytest.param(
+                """
+        fname = "Daniel"
+        age = 12
+        "My name is %s, I'm %s" % (fname, age)
+        """,
+                id="empty-indexes-from-positional",
+            ),
+            pytest.param(
+                """
+        fname = "Daniel"
+        age = 12
+        "My name is %0s, I'm %1s" % (fname, age)
+        """,
+                id="numbered-indexes-from-positionl",
+            ),
+            pytest.param(
+                """
+        fname = "Daniel"
+        age = 12
+        "My name is %(fname)s, I'm %(age)s" % {"fname": fname, "age": age}
+        """,
+                id="named-indexes-from-keyword",
+            ),
+            pytest.param(
+                """
+        string = "My name is %s, I'm %s"
+        string % ("Daniel", 12)
+        """,
+                id="empty-indexes-on-variable",
+            ),
+            pytest.param(
+                """"My name is Daniel, I'm %s" % 12""", id="empty-indexes-from-variable"
+            ),
+            pytest.param(
+                """
+                age = 12
+                "My name is Daniel, I'm %s" % age
+                """,
+                id="empty-indexes-from-variable",
+            ),
+        ],
+    )
+    def test_string_modulation(self, format_string: str) -> None:
+        node: nodes.Call = _extract_single_node(format_string)
+        inferred = next(node.infer())
+        assert isinstance(inferred, nodes.Const)
+        assert inferred.value == "My name is Daniel, I'm 12"
+
+    @pytest.mark.parametrize(
+        "format_string",
+        [
+            """
+            from missing import Unknown
+            fname = Unknown
+            age = 12
+            "My name is %(fname)s, I'm %(age)s" % {"fname": fname, "age": age}
+            """,
+            """
+            from missing import fname
+            age = 12
+            "My name is %(fname)s, I'm %(age)s" % {"fname": fname, "age": age}
+            """,
+            """
+            from missing import fname
+            "My name is %s, I'm %s" % (fname, 12)
+            """,
+            """
+            "My name is %0s, I'm %1s" % ("Daniel")
+            """,
+            """"I am %s" % ()""",
+            """"I am %s" % i for i in (12,)""",
+            """
+            fsname = "Daniel"
+            "My name is %(fname)s, I'm %(age)s" % {"fsname": fsname, "age": age}
+            """,
+            """
+            fname = "Daniel"
+            age = 12
+            "My name is %0s, I'm %(age)s" % (fname, age)
+            """,
+        ],
+    )
+    def test_string_modulation_uninferable(self, format_string: str) -> None:
```

Comment:
Same

Reply:
I can't use `%` in a function name. So `percentage_formatting` then?

Reply:
How about ``c_style_formatting`` ``old_string_formatting`` or ``interpolation_formatting`` ? (I checked the python doc to see how it's called : https://docs.python.org/3/tutorial/inputoutput.html#old-string-formatting)